README.md

```diff
@@ -1,12 +1,106 @@
+```
+
+Note that this project runs under an MIT license and we only permit the use of non-copyleft-licensed packages. Please be aware of this when installing new packages and inform yourself before blindly installing.
+
+When you have any issues with the environment contact `felix-zailskas`.
+
+## Contribution Workflow
+
+### Branching Strategy
 
 **main**: It contains fully stable production code
-* **dev**: It contains stable under-development code
-    
-  * **epic**: It contains a module branch. Like high level of feature. For example, we have an authentication module then we can create a branch like "epic/authentication"
 
-    * **feature**: It contains specific features under the module. For example, under authentication, we have a feature called registration. Sample branch name: "feature/registration"
+- **dev**: It contains stable under-development code
+
+  - **epic**: It contains a module branch. Like high level of feature. For example, we have an authentication module then we can create a branch like "epic/authentication"
+
+    - **feature**: It contains specific features under the module. For example, under authentication, we have a feature called registration. Sample branch name: "feature/registration"
+
+    - **bugfix**: It contains bug fixing during the testing phase and branch name start with the issue number for example "bugfix/3-validate-for-wrong-user-name"
+
+### Commits and Pull Requests
+
+The stable branches `main` and `dev` are protected against direct pushes. To commit code to these branches create a pull request (PR) describing the feature/bugfix that you are committing to the `dev` branch. This PR will then be reviewed by another SD from the project. Only after being approved by another SD a PR may be merged into the `dev` branch. Periodically the stable code on the `dev` branch will be merged into the `main` branch by creating a PR from `dev`. Hence, every feature that should be committed to the `main` branch mus first run without issues on the `dev` branch for some time.
+
+Before contributing to this repository make sure that you are identifiable in your git user settings. This way commits and PRs created by you can be identified and easily traced back.
+
+```[bash]
+git config --local user.name "Manu Musterperson"
+git config --local user.email "manu@musterperson.org"
+```
+
+Any commit should always contain a commit message that references an issue created in the project. Also, always signoff on your commits for identification reasons.
+
+```[bash]
+git commit -m "Fixed issue #123" --signoff
+```
+
+When doing pair programming be sure to always have all SDs mentioned in the commit message. Each SD should be listed on a new line for clarity reasons.
+
+```[bash]
+git commit -a -m "Fixed problem #123
+> Co-authored-by: Manu Musterperson <manu.musterperson@fau.de>"
+```
+
+### License
+
+This project is operated under an MIT license. Every file must contain the REUSE-compliant license and copyright declaration:
```

Comment:
What about binary files?

Reply:
Also, can you add a link here with more information regarding REUSE and SPDX?

Reply:
Check the REUSE documentation for this. https://reuse.software/tutorial/#binary-and-uncommentable-files

Reply:
"You may choose to drop items except the copyright holder, which must always be included. We recommend that you include all items, however." So I am afraid that we need to add a copyright holder to all files as well.

Reply:
I am aware of that but unsure who will be the copyright holder. I will ask that in the lecture and add it afterwards.

Reply:
I think some link (maybe https://reuse.software/faq/) should be in the README for future reference. Can you do that? I don't want to run into merge conflicts with you :-)

Reply:
I think the copyright holder will be all of us / our team. So maybe we can just put "Sum Insights" there? But it definitely won't do any harm to ask in the lecture, thanks for that.

Reply:
The copyright holder are most likely the authors of the files. Berkay and I have added them as the holders for now and I will ask again in the lecture.